compiler/rustc_middle/src/mir/interpret/value.rs

```diff
@@ -37,6 +37,9 @@ pub enum ConstValue<'tcx> {
     /// Used only for `&[u8]` and `&str`
     Slice { data: ConstAllocation<'tcx>, start: usize, end: usize },
 
+    /// Like `Slice`, but for types that aren't 1 byte long.
+    CustomSlice { data: ConstAllocation<'tcx>, length: usize },
```

Comment:
Unclear to me why `Slice` cannot be adapted for this, but my opinion here is probably not worth much since I don’t fiddle much around `mir/interpret`.